igvm/vm.py

```diff
@@ -335,6 +469,95 @@ def build(self, run_puppet=True, debug_puppet=False, postboot=None):
 
         log.info('"{}" is successfully built.'.format(self.fqdn))
 
+    def aws_build(self,
+                  run_puppet: bool = True,
+                  debug_puppet: bool = False,
+                  postboot: str = None,
+                  timeout_vm_setup: int = 300,
+                  timeout_cloud_init: int = 600):
+        """AWS build
+
+        Build a VM in AWS.
+
+        :param: run_puppet: Run puppet (incl. cert clean) after VM creation
+        :param: debug_puppet: Run puppet in debug mode
+        :param: postboot: cloudinit configuration put as userdata
+        :param: timeout_vm_setup: Timeout value for the VM creation
+        :param: timeout_cloud_init: Timeout value for the cloudinit
+                                    provisioning
+        """
+
+        ec2 = boto3.client('ec2')
+
+        try:
+            response = ec2.run_instances(
+                ImageId=self.dataset_obj['aws_image_id'],
+                InstanceType=self.dataset_obj['aws_instance_type'],
+                KeyName=self.dataset_obj['aws_key_name'],
+                SecurityGroupIds=list(
+                    self.dataset_obj['aws_security_group_ids']),
+                SubnetId=self.dataset_obj['aws_subnet_id'],
+                Placement={
+                    'AvailabilityZone': self.dataset_obj['aws_placement']
+                },
+                PrivateIpAddress=str(self.dataset_obj['intern_ip']),
+                UserData='' if postboot is None else postboot,
+                TagSpecifications=[
+                    {
+                        'ResourceType': 'instance',
+                        'Tags': [
+                            {
+                                'Key': 'Name',
+                                'Value': self.dataset_obj['hostname'],
+                            },
+                        ]
+                    },
+                ],
+                DryRun=False,
+                MinCount=1,
+                MaxCount=1)
+            log.info(response)
+        except ClientError as e:
+            raise ClientError(e)
```

Comment:
Shouldn't this be casted to igvm exception?

Reply:
VMError is now raised